src/main/java/org/opensearch/knn/plugin/KNNPlugin.java

```diff
@@ -250,6 +251,12 @@ public Collection<Object> createComponents(
 
         clusterService.addListener(TrainingJobClusterStateListener.getInstance());
 
+        // Initialize k-NN enhanced script service for automatic Mustache support
+        knnScriptService = new KNNScriptService(scriptService);
```

Comment:
Can we remove this logic from create components and just have it all in getScriptEngine?